torch/fx/experimental/proxy_tensor.py

```diff
@@ -172,7 +184,7 @@ def wrapped(*args):
         for idx, arg in enumerate(flat_args):
             if isinstance(flat_inps[idx], torch.Tensor):
                 with no_dispatch():
-                    flat_args[idx] = ProxyTensor(flat_inps[idx], arg)
+                    flat_args[idx] = ProxyTensor(flat_inps[idx], arg, requires_grad=flat_inps[idx].is_leaf)
```

Comment:
This errors out with tensor that is leaf but does not require gradient, e.g. int or boolean tensor.

```
RuntimeError: Only Tensors of floating point and complex dtype can require gradients
```